### DISCUSSION TITLE
Is NET_TCP_BACKLOG_SIZE still used?

### DISCUSSION THREAD

**Discussion by AndreyDodonov-EH** [zephyrproject-rtos/zephyr#44973] [Q&A]
According to the quick grep, it's not used anymore, but still present in KConfig:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/Kconfig#L366

Is it not needed anymore or am I missing something?

**Comment by jukkar** [comment]
Just did a quick check and it is still used
```
zephyr$ (master %=)$ ack BACKLOG_SIZE subsys/net/
subsys/net/ip/tcp.c
55:} tcp_backlog[CONFIG_NET_TCP_BACKLOG_SIZE];
1657:	for (i = 0; i < CONFIG_NET_TCP_BACKLOG_SIZE; i++) {
1864:	for (i = 0; i < CONFIG_NET_TCP_BACKLOG_SIZE; i++) {

subsys/net/ip/Kconfig
302:config NET_TCP_BACKLOG_SIZE
```

**Reply by AndreyDodonov-EH** [reply]
Does your master point to the same commit as main? 
I'm on 83c79d10510e28e8706581168d257c1b556732d2 and have:

```
$ ack BACKLOG_SIZE subsys/net/
subsys/net/ip/Kconfig
357:config NET_TCP_BACKLOG_SIZE
```

**Reply by jukkar** [reply]
Indeed, I had an very old tree I was looking at. After some digging in the git history, the `CONFIG_NET_TCP_BACKLOG_SIZE` disappeared when we switched to new TCP implementation and there is now orphaned config entry as you noticed.
You could send a PR that removes that entry if you wish. If you are adventurous, patches adding support for the config option are very welcome.

**Reply by AndreyDodonov-EH** [reply]
Thanks for confirming!
According to current description, it is support for multiple outgoing connections https://docs.zephyrproject.org/3.0.0/reference/kconfig/CONFIG_NET_TCP_BACKLOG_SIZE.html

While it sounds relevant,
I think there are some higher priority issues and missing features in TCP.
I'll open correspondent issues (hopefully also PR) this or next week and add links to them here for references.

And for this one probably it makes sense to open an issue that it's currently not supported.

**Reply by AndreyDodonov-EH** [reply]
As promised, linking here just for reference, a fresh TCP issue which was a showstopper for me:
https://github.com/zephyrproject-rtos/zephyr/issues/45426
